src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -180,24 +183,49 @@ def __init__(
         if not self.tokenizer.pad_token:
             self.tokenizer.pad_token = self.tokenizer.eos_token
 
-        self.engine = None
-
-        self.multitoken_engine = NLDecoderEngine(
-            onnx_file_path=self.onnx_file_path,
-            engine_type=self.engine_type,
-            engine_args=self.engine_args,
-            engine_context=self.context,
-            sampling_temperature=self.sampling_temperature,
-            deterministic=self.deterministic,
-            sequence_length=self.sequence_length,
-            input_ids_length=prompt_processing_sequence_length,
-            tokenizer=self.tokenizer,
-            use_deepsparse_cache=use_deepsparse_cache,
-        )
+        self.engine, self.multitoken_engine = None, None
+
+        """
+        Inititalize multitoken engine only if:
```

Comment:
additionally, __init__ is getting quite long, can we break some out into helper functions?